mlops/Experiment.py

```diff
@@ -70,59 +66,69 @@ def print_experiment_info(self):
         print("Lifecycle_stage: {}".format(experiment.lifecycle_stage))
 
     def configure_minio(self):
-        uri_formatted = self.config['server']['MLFLOW_S3_ENDPOINT_URL'].replace("http://", "")
-        user = self.config['user']['AWS_ACCESS_KEY_ID']
-        password = self.config['user']['AWS_SECRET_ACCESS_KEY']
-        client = Minio(uri_formatted, user, password, secure=False)
+        if self.use_localhost:
+            self.uri_formatted = self.config['server']['LOCAL_MLFLOW_S3_ENDPOINT_URL'].replace("http://", "")
+        else:
+            self.uri_formatted = self.config['server']['MLFLOW_S3_ENDPOINT_URL'].replace("http://", "")
+
+        self.minio_cred = {'user': os.getenv('MINIO_ROOT_USER'),
+                           'password': os.getenv('MINIO_ROOT_PASSWORD')}
+
+        client = Minio(self.uri_formatted, self.minio_cred['user'], self.minio_cred['password'], secure=False)
         # if mlflow bucket does not exist, create it
         if 'mlflow' not in (bucket.name for bucket in client.list_buckets()):
             print('Creating S3 bucket ''mlflow''')
             client.make_bucket("mlflow")
 
     def build_experiment_image(self, path: str = '.'):
         print('Building experiment image ...')
-        buildargs = {}
-        buildargs['HTTP_PROXY'] = os.getenv('HTTP_PROXY')
-        buildargs['HTTPS_PROXY'] = os.getenv('HTTPS_PROXY')
+
+        # Collect proxy settings
+        build_args = {}
+        if os.getenv('HTTP_PROXY') is not None or os.getenv('HTTPS_PROXY') is not None:
+            build_args = {'HTTP_PROXY': os.getenv('HTTP_PROXY'),
+                          'HTTPS_PROXY': os.getenv('HTTPS_PROXY')}
 
         client = docker.from_env()
-        client.images.build(path=path, tag=self.experiment_name, buildargs=buildargs)
+        client.images.build(path=path,
+                            tag=self.experiment_name,
+                            buildargs=build_args,
+                            rm=True)
+        print('Built: ' + self.experiment_name + ':latest')
 
-    def build_project_file(self):
+    def build_project_file(self, path: str = '.'):
         print('Building project file')
-        projectfile = ProjectFile(self.config)
+        projectfile = ProjectFile(self.config, path=path, use_localhost=self.use_localhost)
         projectfile.generate_yaml()
 
-    def run(self, remote: str = None, **kwargs):
-        num_retries = 1
```

Comment:
Why do we no longer need this retry logic?

Reply:
Same as comment below - improved the error handling method here